packages/tangle/tipmanager.go

```diff
@@ -235,20 +237,238 @@ func (t *TipManager) Tips(p payload.Payload, countParents int) (parents MessageI
 		transaction := p.(*ledgerstate.Transaction)
 
 		tries := 5
-		for !t.tangle.Utils.AllTransactionsApprovedByMessages(transaction.ReferencedTransactionIDs(), parents...) {
+		for !t.tangle.Utils.AllTransactionsApprovedByMessages(transaction.ReferencedTransactionIDs(), parents...) || len(parents) == 0 {
 			if tries == 0 {
-				err = errors.Errorf("not able to make sure that all inputs are in the past cone of selected tips")
+				err = errors.Errorf("not able to make sure that all inputs are in the past cone of selected tips and parents have correct time-since-confirmation")
 				return nil, err
 			}
 			tries--
 
 			parents = t.selectTips(p, MaxParentsCount)
 		}
+	} else {
+		tries := 5
+		for len(parents) == 0 {
+			if tries == 0 {
+				err = errors.Errorf("not able to make sure that parents have correct time-since-confirmation")
+				return nil, err
+			}
+			tries--
+
+			parents = t.selectTips(p, MaxParentsCount)
+		}
+	}
+	return parents, nil
+}
+
+func (t *TipManager) isPastConeTimestampCorrect(messageID MessageID) (timestampValid bool) {
+	now := clock.SyncedTime()
+	markersVisited := 0
+	messagesVisited := 0
+	minSupportedTimestamp := now.Add(-5 * time.Minute)
+	timestampValid = true
+
+	if t.tangle.TimeManager.LastConfirmedMessage().MessageID == EmptyMessageID {
+		// if the genesis message is the last confirmed message, then there is no point in performing tangle walk
+		// return true so that the network can start issuing messages when the tangle starts
+		return
+	}
+
+	if t.tangle.TimeManager.LastConfirmedMessage().Time.Before(minSupportedTimestamp) {
```

Comment:
Yes, with current tangle time implementation this means exactly that, or this could occur during syncing as well. . 
This could rather trigger listening mode (when node is out-of-sync), in which the node can't issue messages and APIs are not fully functional. Then TSC could disable issuing new messages automatically. We are working on it in the tangle time group so there will probably be some changes to it.